pkg/logging/constants.go

```diff
@@ -92,3 +94,31 @@ func FromPrefix(prefix string, obj Object) []interface{} {
 		gvk.GroupVersion(),
 	}
 }
+
+// WithUser adds user identifiers to the logger.
+func WithUser(logger logr.Logger, u user.Info) logr.Logger {
+	return logger.WithValues(FromUser(u)...)
+}
+
+// FromUser provides the structured logging fields that identify a user, prefixing with 'user'.
+func FromUser(u user.Info) []interface{} {
+	return FromUserPrefix(u, "user")
+}
+
+// FromUserPrefix provides the structured logging fields that identify a user, allowing any prefix.
+func FromUserPrefix(u user.Info, prefix string) []interface{} {
+	return []interface{}{
+		prefix + ".name", u.GetName(),
```

Comment:
why not to use camelCase? (like WithCluster)

Reply:
We've been using the period delimiter for fields of one object - we can update `WithCluster` to do that, too, no real strong feeling there. That seemed like more of a holder object for semi-unrelated data (e.g. cluster name requested and whether or not it's a partial metadata request, as opposed to e.g. the name/namespace of an object or name/uid of a user)